addon/components/ember-list.js

```diff
@@ -117,18 +151,20 @@ export default Ember.Component.extend({
 
     for (i=0; i<this.cells.length; i++) {
       cell = this.cells[i];
-      if (!cellMap[cell.index]) {
+      if (!cellMap[cell.key]) {
         if (newItems.length) {
           itemIndex = newItems.pop();
+          itemKey = decodeEachKey(items[itemIndex], '@identity');
           pos = this.cellLayout.positionAt(itemIndex, this.width, this.height);
           width = this.cellLayout.widthAt(itemIndex, this.width, this.height);
           height = this.cellLayout.heightAt(itemIndex, this.width, this.height);
           style = formatStyle(pos, width, height);
           Ember.set(cell, 'style', style);
+          Ember.set(cell, 'key', itemKey);
```

Comment:
I removed the `key="key"` from the `list-view` comonents `{{each}}` so it should be using the `@identity` of the cell and therefore reuse correct?